controllers/admin/AdminAdyenOfficialPrestashopValidatorController.php

```diff
@@ -0,0 +1,286 @@
+    {
+        $invalidConfigs = array();
+        foreach (AdyenOfficial::getAdyenConfigNames() as $key) {
+            if (!Configuration::hasKey($key, null, null, $shopId)) {
+                $invalidConfigs[] = $key;
+                $this->logger->error(sprintf('%s configuration not found in ps_configuration table', $key));
+            }
+        }
+
+        return empty($invalidConfigs);
+    }
+
+    /**
+     * Validate that all adyen module tables have been created in the db
+     *
+     * @return bool
+     */
+    private function validateModuleTables()
+    {
+        $notificationTable = $this->helperData->tableExists('adyen_notification');
+        $responseTable = $this->helperData->tableExists('adyen_payment_response');
+
+        return $notificationTable && $responseTable;
+    }
+
+    /**
+     * Validate that all hooks required by the module are correctly registered
+     *
+     * @return bool
+     */
+    private function validateModuleHooks()
+    {
+        $invalidHooks = array();
+        $allAdyenHooks = AdyenOfficial::getAdyenHooks();
+        if ($this->versionChecker->isPrestaShop16()) {
+            $adyenHooks = $allAdyenHooks['1.6'];
+        } else {
+            $adyenHooks = $allAdyenHooks['1.7'];
+        }
+
+        $moduleHooks = $this->getPossibleHooksList();
+
+        foreach ($adyenHooks as $adyenHook) {
+            $registeredAdyenHook = array_filter($moduleHooks, function ($moduleHook) use ($adyenHook) {
+                return $moduleHook['name'] === $adyenHook && $moduleHook['registered'] === true;
+            });
+            if (empty($registeredAdyenHook)) {
+                $invalidHooks[] = $adyenHook;
+                $this->logger->error(sprintf('%s hook is not registered', $adyenHook));
+            }
+        }
+
+        return empty($invalidHooks);
+    }
+
+    /**
+     * Validate that all order states required by the module have been added
+     *
+     * @param int|false $shopId
+     * @return bool
+     */
+    private function validateOrderStates($shopId)
+    {
+        $invalidOrderStates = array();
+        foreach (AdyenOfficial::getAdyenOrderStates() as $orderStateName) {
+            $orderStateConfigurationId = Configuration::get($orderStateName, null, null, $shopId);
+            $orderState = $this->orderStateAdapter->getOrderStateById($orderStateConfigurationId);
+            if (empty($orderState)) {
+                $invalidOrderStates[] = $orderStateName;
+                $this->logger->error(sprintf('%s order state not found', $orderStateName));
+            }
+        }
+
+        return empty($invalidOrderStates);
+    }
+
+    /**
+     * Get the url accessed when the button is clicked, to download the zip file
+     *
+     * @return string
+     */
+    private function getValidateUrl()
+    {
+        $adminUrl = Tools::getAdminUrl('admin-dev/index.php?controller=AdminAdyenOfficialPrestashopValidator&token=');
```

Comment:
Is admin-dev variable and customizable? could it be different for some merchants?

Reply:
Updated in https://github.com/Adyen/adyen-prestashop/pull/189/commits/d615c4512edabd35e5c5aad1829109242d0f72cf